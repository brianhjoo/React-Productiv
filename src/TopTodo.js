import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ todos }) {
  // lowest-priority # is the highest priority
  let top = todos.reduce(
      (acc, cur) => cur.priority < acc.priority ? cur : acc, todos[0]);

  return (
    <div>
        { todos.length ? (
          <Todo 
          id={top.id} 
          title={top.title} 
          priority={top.priority} 
          description={top.description}
        />) : "No todos yet!" }
    </div>
  )
}

export default TopTodo;