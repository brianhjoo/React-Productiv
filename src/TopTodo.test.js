import React from "react";
import { render } from "@testing-library/react";
import TopTodo from './TopTodo';

const todo1 = {
    id: 1,
    title: "important todo",
    description: "very important",
    priority: 1
}
const todo2 = {
    id: 2,
    title: "not important todo",
    description: "not important",
    priority: 3
}

const todos = [todo1, todo2];

const emptyTodo = [];


describe ("TopTodo Tests", function() {
    it('renders without crashing', function() {
      render(<TopTodo todos={todos}/>);
    });
  
    it('displays the correct TopTodo', function() {
      const result = render(<TopTodo todos={todos} />);
      expect(result.queryByText('very important')).toBeInTheDocument();
      expect(result.queryByText('not important')).not.toBeInTheDocument();
    });

    it('displays the correct message if empty todos', function() {
      const result = render(<TopTodo todos={emptyTodo} />);
      expect(result.queryByText('No todos yet!')).toBeInTheDocument();
    });
  
    it('matches snapshot', function() {
      const { container } = render(<TopTodo todos={todos}/>);
      expect(container).toMatchSnapshot();
    });
});