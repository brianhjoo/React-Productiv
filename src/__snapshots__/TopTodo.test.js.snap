// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo Tests matches snapshot 1`] = `
<div>
  <div
    class="TopTodo"
  >
    <div
      class="Todo"
    >
      <div>
        <b>
          important todo
        </b>
         
        <small>
          (priority: 
          1
          )
        </small>
      </div>
      <div>
        <small>
          very important
        </small>
      </div>
    </div>
  </div>
</div>
`;
